lib/OpenQA/Schema/Result/Jobs.pm

```diff
@@ -676,10 +677,16 @@ sub calculate_result($) {
                 $important_overall = FAILED;
             }
             else {
-                $overall = FAILED;
+                $unimportant_fails = 1;
+                $overall           = FAILED;
             }
         }
     }
+    # override important if it's passed
+    if ($unimportant_fails && $important_overall && $important_overall eq PASSED) {
+        $important_overall = SOFTFAILED;
+    }
+
```

Comment:
which begs the question what happens if overall is SOFTFAILED and important_overall is PASSED